### PR TITLE
Notifications and swipe pages

### PR DESCRIPTION
**Pull request by petarLits**

### CODE REVIEW

File: lib/data/firebase_message_manager.dart [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    }
+    return false;
+  }
+
+  Future<void> initializeLocalNotification() async {
+    await flutterLocalNotificationsPlugin.initialize(
+        InitializationSettings(
+            android: AndroidInitializationSettings('@drawable/background')),
+        onDidReceiveNotificationResponse: _onDidReceiveNotificationResponse);
+  }
```
Comment (UrosLITS) [inline lib/data/firebase_message_manager.dart:96]: you can divide LocalFlutter notifications from FirebaseMessaging that's 2 different packages.

File: lib/home/view/home_page.dart [hunk 2]
```diff
@@ -204,9 +229,45 @@ class _HomePageSate extends State<HomePage> {
     return result;
   }
 
+  void onNotificationListener(String? payload) async {
+    if (payload != null) {
+      final data = jsonDecode(payload);
+      if (data['action'] == 'bookChanged') {
```
Comment (UrosLITS) [inline lib/home/view/home_page.dart]: can put constant here

**Review by UrosLITS — CHANGES_REQUESTED** [review]

File: lib/book/book_page/book_page_view.dart [hunk 3]
```diff
@@ -58,10 +67,15 @@ class _BookPageViewState extends State<BookPageView> {
...
           context.read<BookPageBloc>().add(InitBookEvent(book: widget.book));
         } else if (state is DisplayBookPageState) {
-          widget.book.bookData!.bookPages = state.bookData.bookPages;
+          widget.book.bookData = state.bookData;
           pages = widget.book.bookData!.bookPages;
-          currentPageIndex = state.pageIndex;
           chapters = widget.book.bookData!.bookChapters;
+          if (isMessageReceived) {
+            currentPageIndex = widget.pageIndex!;
+            isMessageReceived = false;
```
Comment (milicaLits) [inline lib/book/book_page/book_page_view.dart:75]: Instead of using flag use getter for isMessageReceived

**Review by milicaLits — CHANGES_REQUESTED** [review]

**Review by milicaLits — APPROVED** [review]

File: lib/home/bloc/home_event.dart [hunk 4]
```diff
@@ -11,12 +12,18 @@ sealed class HomeEvent extends Equatable {
...
+    required this.messageBody,
+  });
 
   final Book book;
+  final String messageTitle;
+  final String messageBody;
 
   @override
-  List<Object?> get props => [book];
+  List<Object?> get props => [book, messageTitle, messageBody];
```
Comment (UrosLITS) [inline lib/home/bloc/home_event.dart:26]: add comma here

**Review by UrosLITS — APPROVED** [review]

File: lib/home/bloc/home_event.dart (hunk 4 above)
Comment (petarLits) [reply lib/home/bloc/home_event.dart:26]: it will be fixed in next pull request

**Pull request merged**